### PR TITLE
Pull request for project review

### PR DESCRIPTION
**Pull request by AjroudiRida**
In this project we did the following tasks:

- Set up Kanban board
- Create app and install basic dependencies - Setup (group task)
- Create website header with empty nav - Setup (group task)
- Create empty My profile - Setup (group task)
- Create basic structure for Rockets - Setup
- Create basic structure for Missions - Setup
- Create basic structure for Dragons - Setup
- Fetch missions - Fetch data
- Fetch rockets - Fetch data
- Display missions - Lists render
- Display rockets - Lists render
- Implement rocket booking - Actions
- Implement mission joining - Actions
- Implement mission leaving - Actions
- Implement rocket booking cancelation - Actions
- Switch badges for Missions - Conditional components
- Switch badges for Rockets - Conditional components
- Display reserved rockets in My profile - Filters
- Display joined missions in My profile - Filters

### CODE REVIEW

File: src/redux/features/missions/missionsSlice.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      const newState = state.missions.map((mission) => {
+        if (mission.mission_id !== payload) {
+          return mission;
+        }
+        return { ...mission, reserved: true };
+      });
+
+      return { ...state, missions: newState };
+      // console.log(payload)
+    },
```
Comment (tresorsawasawa) [inline src/redux/features/missions/missionsSlice.js:38]: - [ ] For the best practices and clean code, kindly ensure that you remove all the commented code lines in your code base.

File: src/setupTests.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+// jest-dom adds custom jest matchers for asserting on DOM nodes.
+// allows you to do things like:
+// expect(element).toHaveTextContent(/react/i)
+// learn more: https://github.com/testing-library/jest-dom
+import '@testing-library/jest-dom';
```
Comment (tresorsawasawa) [inline src/setupTests.js:5]: - [ ] As required for the project to  add tests for all your components and functions, kindly ensure that you create all possible tests to your project. as right now no test has been implemented.

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @AjroudiRida and @ngalamac  
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Used React Router :heavy_check_mark: 
- Used React components :heavy_check_mark: 
- Used store, actions and reducers in React :heavy_check_mark:
- Used duck pattern structure for your files :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Efiamotu-1 — APPROVED** [review]
Hi @AjroudiRida  & @ngalamac , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# STATUS: APPROVED 🕺🥳 

image

## Highlights :dart: 
* Correct Gitflow :heavy_check_mark: 
* Good job implementing the tests for various components ✔️ 
* Good commit messages :heavy_check_mark: 
* All linters passing :heavy_check_mark: 
* Good PR summary :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, you can now Merge since it is Approved ✅, However if there are any conflicts please resolve them before merging.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**